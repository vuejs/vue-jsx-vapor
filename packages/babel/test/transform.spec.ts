import { transformSync } from '@babel/core'
import { describe, expect, test } from 'vitest'
import jsx from '../src/index'

describe('transform', () => {
  test('transform multiple components', () => {
    const { code } = transformSync(
      `const a = <div onClick={onClick}>{Hello}</div>
       const b = <>{foo? <div onClick={onClick}>Hello</div> : <div onDblclick={onDblclick}>World</div>}</>`,
      {
        filename: 'test.tsx',
        plugins: [[jsx]],
      },
    )!
    expect(code).matchSnapshot()
  })
})
