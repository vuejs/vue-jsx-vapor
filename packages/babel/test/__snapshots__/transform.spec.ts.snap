// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transform > transform multiple components 1`] = `
"import { child as _child, delegateEvents as _delegateEvents, template as _template, createIf as _createIf } from 'vue';
import { setNodes as _setNodes } from 'vue-jsx-vapor/runtime';
const t0 = _template("<div> </div>", true),
  t1 = _template("<div>Hello</div>"),
  t2 = _template("<div>World</div>");
_delegateEvents("click", "dblclick");
const a = (() => {
  const n0 = t0();
  const x0 = _child(n0);
  _setNodes(x0, () => Hello);
  n0.$evtclick = e => onClick(e);
  return n0;
})();
const b = (() => {
  const n0 = _createIf(() => foo, () => {
    const n2 = t1();
    n2.$evtclick = e => onClick(e);
    return n2;
  }, () => {
    const n4 = t2();
    n4.$evtdblclick = e => onDblclick(e);
    return n4;
  });
  return n0;
})();"
`;
